evalml/automl/automl_search.py

```diff
@@ -260,6 +262,10 @@ def __init__(self,
         except ImportError:
             logger.warning("Unable to import plotly; skipping pipeline search plotting\n")
 
+        if allowed_pipelines is not None and not isinstance(allowed_pipelines, list):
```

Comment:
Adding this data validation so that custom_hyperparameters can be overloaded with `pipeline_parameters` without weird bugs.